src/test/java/com/ea/restaurant/service/impl/UserServiceImplTest.java

```diff
@@ -0,0 +1,91 @@
+package com.ea.restaurant.service.impl;
+
+import com.ea.restaurant.constants.Status;
+import com.ea.restaurant.entities.User;
+import com.ea.restaurant.repository.UserRepository;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+class UserServiceImplTest {
+
+  @Mock UserRepository userRepository;
+
+  @InjectMocks UserServiceImpl userService;
+
+  @BeforeEach
```

Comment:
Does `@Before` works?

Reply:
`@beforeEach` is the equivalent of `@before` in Junit5

Reply:
Sounds good!

Reply:
Use `@BeforeAll`

Reply:
In this case, i prefer use `@beforeEach` as beforeAll create an instance of User service at the beginning and my verify check how many times a method was call, then i will count methods call in test running before.